SU2_CFD/include/fluid/CFluidModel.hpp

```diff
@@ -339,6 +355,13 @@ class CFluidModel {
    */
   virtual void SetTDState_T(su2double val_Temperature, const su2double* val_scalars = nullptr) {}
 
+  /*!
+   * \brief Virtual member.
+   * \param[in] val_enthalpy - Enthalpy value at the point.
+   */
+  virtual void ComputeTempFromEnthalpy(su2double val_enthalpy, su2double* val_temperature,
+                                       const su2double* val_scalars = nullptr) {}
```

Comment:
This seems inconsistent with the SetTDState_* functions.
Should this be SetTDState_h and then you use GetTemperature to get the temperature?

Reply:
In the pull request made by evert regarding preferential diffusion, he also used the same function for the flamelet solver. The only difference is that  that function is in the CSpeciesFlameletSolver.cpp and I cannot access to that function where I set the primitive in SetPrimVar in CIncNSVariable.cpp

Reply:
My point is again to keep an abstract interface.
Evert's function is a private function of a specific solver which is perfectly fine, and it uses the fluid model... If you need a similar function, move it somewhere higher up in the hierarchy to make it accessible. For example the scalar solver. Or even the fluid model but implemented in a way that keeps it useful for all fluids.